File: packages/player/android/src/main/kotlin/br/com/suamusica/player/MethodChannelManager.kt
package br.com.suamusica.player

import io.flutter.plugin.common.MethodChannel

class MethodChannelManager(private val channel: MethodChannel) {

    fun notifyPositionChange(playerId: String, position: Long, duration: Long) {
        val args = ArgsBuilder()
            .playerId(playerId)
            .position(position)
            .duration(duration)
            .build()

        invokeMethod("audio.onCurrentPosition", args)
    }

    fun notifyPlayerStateChange(playerId: String, state: PlayerState) {
        val args = ArgsBuilder()
            .playerId(playerId)
            .state(state)
            .build()

        invokeMethod("state.change", args)
    }
    fun notifyError(playerId: String, state: PlayerState, error: String? = null) {
        val args = ArgsBuilder()
            .playerId(playerId)
            .state(state)
            .error(error)
            .build()

        invokeMethod("state.change", args)
    }

    fun notifyNext(playerId: String) {
        val args = ArgsBuilder()
            .playerId(playerId)
            .build()
        invokeMethod("commandCenter.onNext", args)
    }
    fun notifyPrevious(playerId: String) {
        val args = ArgsBuilder()
            .playerId(playerId)
            .build()
        invokeMethod("commandCenter.onPrevious", args)
    }
    fun notifyItemTransition(playerId: String) {
        val args = ArgsBuilder()
            .playerId(playerId)
            .state(PlayerState.ITEM_TRANSITION)
            .build()
        invokeMethod("state.change", args)
    }

    private fun invokeMethod(method: String, args: Map<String, Any>) {
        channel.invokeMethod(method, args)
    }

    fun currentMediaIndex(playerId: String, currentMediaIndex: Int) {
        val args = MethodChannelManagerArgsBuilder()
            .playerId(playerId)
            .currentMediaIndex(currentMediaIndex)
            .build()
        invokeMethod("SET_CURRENT_MEDIA_INDEX", args)
    }
    fun onRepeatChanged(playerId: String, repeatMode: Int) {
        val args = MethodChannelManagerArgsBuilder()
            .playerId(playerId)
            .repeatMode(repeatMode)
            .build()
        invokeMethod("REPEAT_CHANGED", args)
    }
    fun onShuffleModeEnabled(playerId: String, shuffleModeEnabled: Boolean) {
        val args = MethodChannelManagerArgsBuilder()
            .playerId(playerId)
            .shuffleModeEnabled(shuffleModeEnabled)
            .build()
        invokeMethod("SHUFFLE_CHANGED", args)
    }

}

File: packages/player/ios/Classes/MethodChannelManager.swift
//
//  MethodChannelManager.swift
//  smplayer
//
//  Created by Lucas Tonussi on 26/09/24.
//



import Foundation
import Flutter.FlutterPlugin



public class MethodChannelManager:NSObject{
    let channel: FlutterMethodChannel?
    init(channel: FlutterMethodChannel?) {
        self.channel = channel
    }
    var tag = "MethodChannelManager"
    
    func notifyPositionChange(
        position: Double,
        duration: Double) {
        let args = MethodChannelManagerArgsBuilder()
            .playerId(id:"smplayer")
            .position(position:Int(position)*1000)
            .duration(duration:Int(duration)*1000)
            .build()
            channel?.invokeMethod("audio.onCurrentPosition", arguments: args)
    }
    
    func notifyNetworkStatus(
    status:Bool) {
        let args = MethodChannelManagerArgsBuilder()
            .playerId(id:"smplayer")
            .build()
        channel?.invokeMethod("network.onChange", arguments: status ? "CONNECTED" : "DISCONNECTED")
    }
    
    
    func notifyPlayerStateChange(state: PlayerState) {
        print("#CheckListeners - notifyPlayerStateChange \(state)")
        let args = MethodChannelManagerArgsBuilder()
            .playerId(id:"smplayer")
            .state(state:state)
            .build()
        channel?.invokeMethod("state.change", arguments: args)
    }
    
    func notifyError(error: String? = nil) {
         let args = MethodChannelManagerArgsBuilder()
            .playerId(id: "smplayer")
            .state(state:PlayerState.error)
            .error(error:error)
            .build()
        channel?.invokeMethod("state.change", arguments: args)
     }
    
    func currentMediaIndex(index: Int) {
        let args = MethodChannelManagerArgsBuilder()
            .playerId(id:"smplayer")
            .currentIndex(index:index)
            .build()
        channel?.invokeMethod(SET_CURRENT_MEDIA_INDEX, arguments: args)
    }

    func repeatModeChanged(repeatMode: Int) {
        let args = MethodChannelManagerArgsBuilder()
            .repeatMode(repeatMode: repeatMode)
            .build()
        channel?.invokeMethod(REPEAT_MODE_CHANGED, arguments: args)
    }
    
    func shuffleChanged(shuffleIsActive: Bool) {
        let args = MethodChannelManagerArgsBuilder()
            .shuffleIsActive(shuffleIsActive: shuffleIsActive)
            .build()
        channel?.invokeMethod(SHUFFLE_CHANGED, arguments: args)
    }
}

